### PR TITLE
added sparse table

### PR DESCRIPTION
**Pull request by chaudhary19**
added range queries using sparse table
1) range minimum query using sparse table
2) range maximum query using sparse table
3) range sum query using sparse table

### CODE REVIEW

**Comment by cclauss** [comment]
Please read CONTRIBUTING.md and add doctests and/or type hints.

**Comment by chaudhary19** [comment]
> Please read CONTRIBUTING.md and add doctests and/or type hints.

But sir I already provide comments along with code of sparse table for all type of queries
please give me any suggestion where should I add comments and hints
thank you
and regards
Mayank Chaudhary

File: data_structures/sparse_table/range_max_query.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            i = 0
+            while (i + (1 << j) - 1) < self.size:
+                if arr[self.stable[i][j - 1]] > arr[self.stable[i + (1 << (j - 1))][j - 1]]:
+                    self.stable[i][j] = self.stable[i][j - 1]
+                else:
+                    self.stable[i][j] = self.stable[i + (1 << (j - 1))][j - 1]
+                i += 1
+            j += 1
+
+    def max_query(self,l ,r):   # Range Maximum Query  :  [l,r)
```
Comment (cclauss) [inline data_structures/sparse_table/range_max_query.py]:
```python
def max_query(self, l: int, r: int):  # Range Maximum Query  :  [l,r)
        """
        >>> sparse_table = SparseTable([1,2,3,4,5,6,7,8,9])
        >>> sparse_table.max_query(2,5)
        6
        >>> sparse_table.max_query(1,8)
        9
        """
```

Comment (chaudhary19) [reply data_structures/sparse_table/range_max_query.py]: thank you sir..I got it how to add them

File: data_structures/sparse_table/range_min_query.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            while (i + (1 << j) - 1) < self.size:
+                if arr[self.stable[i][j - 1]] < arr[self.stable[i + (1 << (j - 1))][j - 1]]:
+                    self.stable[i][j] = self.stable[i][j - 1]
+                else:
+                    self.stable[i][j] = self.stable[i + (1 << (j - 1))][j - 1]
+                i += 1
+            j += 1
+
+    def min_query(self,l ,r):   # Range minimum Query  :  [l,r)
+        length = r-l+1
```
Comment (cclauss) [inline data_structures/sparse_table/range_min_query.py]: Need doctests here.

File: data_structures/sparse_table/range_sum_query.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        j = 1
+        while (1 << j) <= self.size:   # Build the Sparse table
+            i = 0
+            while (i + (1 << j) - 1) < self.size:
+                self.stable[i][j] = self.stable[i][j - 1] + self.stable[i + (1<<(j-1))][j-1]
+                i += 1
+            j += 1
+
+    def sum_query(self,l ,r):   # Range Sum Query  :  [l,r)
```
Comment (cclauss) [inline data_structures/sparse_table/range_sum_query.py]: Need doctests here.

File: data_structures/sparse_table/range_max_query.py [hunk 4]
```diff
@@ -21,16 +27,24 @@ def __init__(self, arr):  # create Sparse Table using given Array
...
-    print(s_obj.max_query(2,5))
-    print(s_obj.max_query(1,8))
-
+    """
+        >>> sparse_table = SparseTable([1, 2, 3, 4, 5, 6, 7, 8, 9])
+        >>> sparse_table.max_query(2, 5)
+        6
+        >>> sparse_table.max_query(1, 8)
+        9
+    """
```
Comment (cclauss) [inline data_structures/sparse_table/range_max_query.py]: Please dedent lines 42-46 so they left aline with 41 and 47.

**Pull request closed without merging**